Fix #560: Create Oppia Android README

BenHenning opened the pull request:
Fix #560.
Fix part of #1723.

Introduce a README for the repository, including badges for our GitHub actions runs & Gitter channel. This README is heavily based on Oppia web/backend's: https://github.com/oppia/oppia/blob/develop/.github/README.md.

BenHenning committed:
Create Oppia Android README

BenHenning committed:
s/Oppia/Oppia Android/ in the title

BenHenning commented:
Note that I'm thinking we can add a few pictures once the alpha content is ready to be loaded into a binary (@rt4914 FYI).

BenHenning commented:
@anandwana001 & @rt4914: do you think this README is sufficient per your notes in #560?

anandwana001 commented on .github/README.md:
```diff
@@ -0,0 +1,47 @@
+# [Oppia Android](https://www.oppia.org) ![CI Lint and Tests](https://github.com/oppia/oppia-android/workflows/CI%20Lint%20and%20Tests/badge.svg) [![Gitter](https://badges.gitter.im/oppia/oppia-android.svg)](https://gitter.im/oppia/oppia-android?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge)
```
1. As per the CI and Lint passing badge, we can add a branch parameter to be more specific that on which branch all the work is going on - `https://github.com/oppia/oppia-android/workflows/CI%20Lint%20and%20Tests/badge.svg?branch=develop`
2. We can show it to the bottom of the title
```
# [Oppia Android](https://www.oppia.org) 
![CI Lint and Tests](https://github.com/oppia/oppia-android/workflows/CI%20Lint%20and%20Tests/badge.svg?branch=develop) [![Gitter](https://badges.gitter.im/oppia/oppia-android.svg)](https://gitter.im/oppia/oppia-android?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge)
```
3. We can add a todo numbers badge which shows that currently how many todos are there in the codebase which might attract new contributors as well as a type of track
4. We can also add a badge showing what version of ktlint, bufbuild, Android Studio, Kotlin, bazel we are working on. This is the first and the foremost thing a new contributor wants to know about before start working contributing so that a contributor must have everything set up properly.   
5. We can add Oppia's social media(YouTube, Medium, Twitter, Mailing List) badges here like: [![Twitter Follow](https://img.shields.io/twitter/follow/oppiaorg.svg?style=social&label=Follow&maxAge=2592000?style=flat-square)](https://twitter.com/oppiaorg) [![Mailing List](https://img.shields.io/badge/Mailing%20oppia-android-dev.svg)](mailto:oppia-android-dev@googlegroups.com)

BenHenning replied:
1. Done.
2. This is intentionally on the same line as the title for consistency with https://github.com/oppia/oppia.
3. Is there an existing library for this? I like the idea of tracking them.
4. Sounds like a good idea--how would we do that?
5. Nice--done.

anandwana001 replied:
3. Please check this - https://www.tickgit.com/
Adding this badge as an example. 
[![TODOs](https://img.shields.io/endpoint?url=https://api.tickgit.com/badge?repo=github.com/cilium/hubble)](https://www.tickgit.com/browse?repo=github.com/cilium/hubble)

4. We can go hardcode for now as we are migrating from gradle to bazel. Most of the solutions accept gradle file from which they extract the version and show it to the badge. 
[![Kotlin version badge](https://img.shields.io/badge/kotlin-1.2.31-9cf.svg)](http://kotlinlang.org/)
[![Ktlint version badge](https://img.shields.io/badge/Ktlint-0.37.0-9cf.svg)]()
[![bufbuild version badge](https://img.shields.io/badge/Bufbuild-v0.15.0-9cf.svg)]()
[![JDK version badge](https://img.shields.io/badge/Java-1.9-9cf.svg)]()
[![Android Studio version badge](https://img.shields.io/badge/Android%20Studio-4.1-9cf.svg)]()
[![Bazel version badge](https://img.shields.io/badge/Bazel-3.4.1-9cf.svg)]()

This is something hacktoberfest worthy to add. 
[![GitHub issues by-label](https://img.shields.io/github/issues/oppia/oppia-android/good%20first%20issue.svg)](https://github.com/oppia/oppia-android/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)

anandwana001 replied:
we can only put effort on point 3 if we have time and you find it worthy, else we are ok to skip it.

BenHenning replied:
Do you see specific value in adding the versions? I'm not as inclined to if we can't do it automatically since theyu'll quickly become out of date.

I like the idea of the TODOs badge, but tickgit.com isn't working for me when I try to add oppia-android. Any ideas?

I like the good first issues badge link. I'll add that.

anandwana001 replied:
I think the version think we can take after we completely remove the Gradle, so in that we way we can have a badge which always gets updated with the required bazel file for version. 

tickgit.com also not working for me, maybe some setup is required. We can add this as a good first issue.

BenHenning replied:
That sounds good to me. Does this look resolved to you then, @anandwana001?

anandwana001 replied:
Yes

anandwana001 commented on .github/README.md:
```diff
@@ -0,0 +1,47 @@
+
+The Oppia project is built by the community for the community. We welcome contributions from everyone, especially new contributors.
+
+You can help with Oppia's development in many ways, including art, coding, design and documentation.
```
We should highlight `art, coding, design and documentation`. Maybe with the links of that particular wiki page, we can add here or we can add the same as we had in our home wiki page.

- Coders
- Instructional designers and storytellers
- UX researchers
- Voice artists
- Designers and artists

BenHenning replied:
Nice idea, done.

anandwana001 commented on .github/README.md:
```diff
@@ -0,0 +1,47 @@
+## Installation
+
+Please refer to the [installation wiki page](https://github.com/oppia/oppia-android/wiki#installation) for full instructions.
+
```
We can have a section **## Development** here where we can describe a brief about the libraries, architecture we have in our codebase. This will bring two things:
1. Attract contributors - Well documented codebase, new learning opportunities, new tech stacks like bazel
2. Knowledge of codebase at first look - data flow, architecture

For this, initially, we can add a brief from - https://github.com/oppia/oppia-android/wiki/Overview-of-the-Oppia-Android-codebase-and-architecture

anandwana001 replied:
As per the brief libraries description, we can add a small list of amazing libraries we are using to support point 1 and 2 in the above comment. 

```
Build System
- Bazel
-Gradle

List of amazing libraries we are using
- databinding
- navigation
- com.chaos.view:pinview
- glide
- retrofit2
- de.hdodenhof:circleimageview
- io.fabric.sdk.android:fabric
- kotlin-reflect
- kotlinx-coroutines-android
- androidx.work
- com.caverock:androidsvg
- espresso
- robolectric
- mockito
- google truth
```

BenHenning replied:
I think this is a nice idea, though I'm limiting the number of libraries we list to avoid overwhelming the reader. PTAL & let me know what you think. This is definitely a section that we can add onto in the future.

anandwana001 replied:
Look perfect

anandwana001 commented on .github/README.md:
```diff
@@ -0,0 +1,47 @@
+
+## License
+
+The Oppia ANdroid code is released under the [Apache v2 license](https://github.com/oppia/oppia-android/blob/develop/LICENSE).
```
~ANdroid~ Android

BenHenning replied:
Good catch, fixed.

anandwana001 reviewed (changes requested):
Put a few of my thoughts here.

rt4914 commented:
@anandwana001 Just assign it back to me once your comments are resolved considering it will change the readme a lot and also most of the things will get convered.

BenHenning committed:
Update README.md

Address review comments.

BenHenning commented:
Thanks @anandwana001! I appreciate your thorough review. PTAL at the latest changes.

anandwana001 commented on .github/README.md:
```diff
@@ -0,0 +1,61 @@
+
+The Oppia project is built by the community for the community. We welcome contributions from everyone, especially new contributors.
+
+You can help with Oppia's development in many ways, including coding, [instructional design & storytelling](https://github.com/oppia/oppia/wiki/Teaching-with-Oppia), [UX research](https://github.com/oppia/oppia/wiki/Conducting-research-with-students), [creating voiceovers](https://github.com/oppia/oppia/wiki/Instructions-for-voice-artists), [design & art](https://github.com/oppia/oppia/wiki/Contr ...
```
How about linking coding too?
https://github.com/oppia/oppia-android/wiki

anandwana001 replied:
How about linking coding too?
https://github.com/oppia/oppia-android/wiki

anandwana001 commented on .github/README.md:
```diff
@@ -0,0 +1,61 @@
+
+The Oppia project is built by the community for the community. We welcome contributions from everyone, especially new contributors.
+
+You can help with Oppia's development in many ways, including coding, [instructional design & storytelling](https://github.com/oppia/oppia/wiki/Teaching-with-Oppia), [UX research](https://github.com/oppia/oppia/wiki/Conducting-research-with-students), [creating voiceovers](https://github.com/oppia/oppia/wiki/Instructions-for-voice-artists), [design & art](https://github.com/oppia/oppia/wiki/Contr ...
```
How about linking coding too?
https://github.com/oppia/oppia-android/wiki

BenHenning replied:
Done (but I used https://github.com/oppia/oppia-android/wiki#instructions-for-making-a-code-change).

anandwana001 reviewed (approved):
LGTM, Thanks @BenHenning This seems amazing.

anandwana001 commented:
Are we adding License to Readme?

BenHenning commented:
Isn't the license being referenced in the ``## License`` section, or am I missing something @anandwana001?

Also, is this ready for me to follow-up on or are you still reviewing it (per the assignments)?

anandwana001 commented:
> Isn't the license being referenced in the `## License` section, or am I missing something @anandwana001?
> 
> Also, is this ready for me to follow-up on or are you still reviewing it (per the assignments)?

Ah, I missed it, got the license now. 
I was checking the ticket thing, but assigning to you now.

BenHenning committed:
Update README.md

Address reviewer comments: add link for coding contribution instructions, and add badge for newcomer issues.

BenHenning commented:
Thanks @anandwana001. Addressed your comments, PTAL.

@rt4914 also PTAL when you get a chance--I would like to know if you have any thoughts on what should go into the README.

rt4914 reviewed (approved):
This looks really good and can easily be verified in next couple of days by new contributors. Thanks.

BenHenning commented:
Thanks. @anandwana001 please confirm final comment thread, then I can merge.

BenHenning commented:
Thanks everyone! Going ahead and merging this now.
